### PR TITLE
feat(auth-ids): Natural delegations always get automatic scopes.

### PR DESCRIPTION
**Pull request by valurefugl**
https://www.notion.so/Look-into-supporting-natural-delegations-without-API-scopes-817e39e4bf4d4a66af17033e3776fd13?pvs=4

## What

Natural delegations get scopes marked as `automaticDelegationGrant` even if they don't have any api scopes.

## Why

So clients not using api scopes can support natural delegations.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  },
+  '2': {
+    fromNationalId: createNationalId('company'),
+    delegationType: [DelegationType.ProcurationHolder],
+    expected: [...procurationHolderScopes, ...identityResources],
+  },
+  '3': {
+    fromNationalId: fromCustom[0].nationalId,
+    delegationType: [DelegationType.Custom],
+    expected: [...customScopes1, ...identityResources],
```
Comment (saevarma) [inline apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts:59]: Could do this instead as you are using `fromCustom`

```suggestion
expected: [...fromCustom[0].scope, ...identityResources],
```

File: apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  },
+  '4': {
+    fromNationalId: fromCustom[0].nationalId,
+    delegationType: [DelegationType.LegalGuardian, DelegationType.Custom],
+    expected: [...legalGuardianScopes, ...customScopes1, ...identityResources],
```
Comment (saevarma) [inline apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts:64]: Same here `fromCustom[0].scope`

File: apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+              toNationalId: user.nationalId,
+              fromNationalId: d.nationalId,
+              scopes: d.scopes.map((s: string) => ({
+                scopeName: s,
+              })),
+            }))
+            .map((delegation) => factory.createCustomDelegation(delegation)),
+        )
+      })
+
```
Comment (saevarma) [inline apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts:159]: It is unnecessary to do two maps, i.e.

```
        await Promise.all(
          apiScopes
            .map((s) => factory.createApiScope({
              name: s,
              description: s,
              displayName: s,
              domainName: domainName,
              grantToLegalGuardians: legalGuardianScopes.includes(s),
              grantToProcuringHolders: procurationHolderScopes.includes(s),
              allowExplicitDelegationGrant: [
                ...customScopes1,
                ...customScopes2,
              ].includes(s),
              automaticDelegationGrant: false,
              grantToPersonalRepresentatives: false,
              isAccessControlled: false,
            }))
        )
```

File: apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+const apiScopes = [
+  ...legalGuardianScopes,
+  ...procurationHolderScopes,
+  ...customScopes1,
+  ...customScopes2,
+]
+
+const fromCustom = [
+  { nationalId: createNationalId('person'), scopes: customScopes1 },
+  { nationalId: createNationalId('person'), scopes: customScopes2 },
```
Comment (saevarma) [inline apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts:35]: Is this used?

File: libs/services/auth/testing/src/fixtures/fixture-factory.ts [hunk 5]
```diff
@@ -68,6 +71,22 @@ export class FixtureFactory {
     return this.get(ClientAllowedScope).create(scope)
   }
 
+  async createIdentityResource({
+    ...identityResource
+  }: CreateIdentityResource = {}): Promise<IdentityResource> {
```
Comment (saevarma) [inline libs/services/auth/testing/src/fixtures/fixture-factory.ts:76]:
```suggestion
  async createIdentityResource(identityResource: CreateIdentityResource = {}): Promise<IdentityResource> {
  ```

File: apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts (hunk 3 above)
Comment (valurefugl) [reply apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts]: Fixed.

File: libs/services/auth/testing/src/fixtures/fixture-factory.ts (hunk 5 above)
Comment (valurefugl) [reply libs/services/auth/testing/src/fixtures/fixture-factory.ts]: Fixed.

File: apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts (hunk 4 above)
Comment (saevarma) [reply apps/services/auth/ids-api/src/app/delegations/test/delegations-scopes.spec.ts:35]: Yes I see.

**Review by saevarma — APPROVED** [review]
LGTM

**Pull request merged**